fabric-enchantments-v1/src/main/java/net/fabricmc/fabric/api/enchantment/v1/EnchantingPowerProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.enchantment.v1;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+/**
+ * To be used on blocks which increase the power of an enchanting table.
+ */
+public interface EnchantingPowerProvider {
+	/**
+	 * Returns how much the enchanting power should be increased by.
+	 *
+	 * @param blockState The blockstate
+	 * @param world The world of the block
+	 * @param blockPos The block position of the block
+	 * @return The power amount provided by this block
```

Comment:
Might be useful putting some refrence values here. Without looking into vanilla I wouldnt know what kind of numbers to return.

Reply:
Vanilla only ever gives one "power unit" per block.

Reply:
Can we support like fractional power etc? Just curious.

Reply:
> Can we support like fractional power etc? Just curious.

Would be possible: Do you want floats or doubles? 😀

Reply:
Hmm, after some thinking we should provide like extra callbacks that actively scan the world and return a highest ench power allowed. vanilla logic will be delegated to the callback (i.e. returning 30 if there is 15+ bookshelves, etc) so that mods can also raise the cap of 30 with their custom blocks, etc. or detect with fractional-power blocks.

Reply:
I'm not sure if that's what you mean but I would then just add up everything that the callbacks return.
This would also mean that we would need to get rid/copy a lot of vanilla code and additionally would need to move the code out of vanilla's loop (which I was hardly punished for when I originally created this pr).

Reply:
Hmm, this is a bit difficult... The mixin's target code is a bit problematic because it is like in a lambda class, which would have like different class names during each run. Ain't sure if mixin can handle that